scripts/check-go-version-yaml.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Function to check if the YAML file contains the specified Go version after
+# field 'go:'.
+check_go_version_yaml() {
+    local yamlfile="$1"
+    local required_go_version="$2"
+
+    # Use grep to find lines with 'go:'. The grep exist status is ignored.
```

Comment:
do we use the prefix `go:`? I thought we always use `GO_VERSION:`

Reply:
in yaml it's often `go: `

Reply:
Good point. I think the only example is in `.golangci.yml`:
```
  staticcheck:
    go: "1.21"
    checks: ["-SA1019"]
 ```
 
 And I don't think even care about that case. See: https://github.com/lightningnetwork/lnd/pull/8954#discussion_r1697101580

Reply:
Ok, in the latest commits, I've bumped the Go version in `.golangci.yml`. And I've modified the lint check so that it's included in the check.